python/chainer_compiler.py

```diff
@@ -180,23 +227,9 @@ def __init__(self, model, inputs, translator='ch2o', dump_onnx=False,
             self.compile(inputs)
 
     def compile(self, inputs):
```

Comment:
As chatted locally, I think it's better to make this feature available only when the user explicitly specify a parameter. Otherwise, end users may be confused by `to_gpu`-ed process consuming a lot of host memory.

Reply:
Added use_unified_memory option.